src/typescript/typescript-parser.ts

```diff
@@ -31,120 +27,15 @@ export class TypeScriptParser implements Parser<ParsedTypeScriptDocument> {
   }
 
   parse(contents: string, url: string): ParsedTypeScriptDocument {
-    const host = new AnalyzerCompilerHost(
-        this._analyzer, new Map([[url, contents], ['lib.ts', '']]));
-    const program = ts.createProgram([url], _compilerOptions, host);
+    const sourceFile =
+        ts.createSourceFile(url, contents, ts.ScriptTarget.ES2016);
     return new ParsedTypeScriptDocument({
       url,
       contents,
-      program,
       locationOffset: undefined,
-      // TODO(justinfagnani): Fix or remove generics on ParsedDocument that
-      // make this 1) necessary 2) possible (not throwing)
-      ast: null as any as ts.Node,
-      astNode: null,
+      ast: sourceFile,
+      astNode: sourceFile,
       isInline: false,
     });
   }
 }
-
```

Comment:
Yeah, I was much happier with this, and re-using SourceFiles in the next PR. Still good that I figured out how to write a CompmilerHost for the language analyzer part.